File: 01/simple.scala
object Simple {
  def simple(a: Int, n: Int): Int = {
    def aux(a: Int, i: Int): Int = {
      if (i > n) a else aux(a + i, i + 1)
    }
    aux(a, 1);
  }

  def main(arguments: Array[String]) {
    println("1 + 2 + ... + 10 = " + simple(0, 10))
  }
}

File: 01/simple.c
#include <stdio.h>

int simple(int a, int n) {
  for (int i = 1; i <= n; i++) {
    a = a + i;
  }
  return a;
}

int main() {
  printf("1 + 2 + ... + 10 = %d\n", simple(0, 10));
}
